DSCResources/MSFT_xWebsite/MSFT_xWebsite.psm1

```diff
@@ -26,53 +26,46 @@ function Get-TargetResource
     [OutputType([System.Collections.Hashtable])]
     param
     (
-        [Parameter(Mandatory)]
+        [Parameter(Mandatory = $true)]
         [ValidateNotNullOrEmpty()]
-        [string]$Name,
+        [String]
+        $Name,
 
-        [Parameter(Mandatory)]
+        [Parameter(Mandatory = $true)]
         [ValidateNotNullOrEmpty()]
-        [string]$PhysicalPath
+        [String]
+        $PhysicalPath
     )
 
     # Check if WebAdministration module is present for IIS cmdlets
-    if(!(Get-Module -ListAvailable -Name WebAdministration))
+    if (-not (Get-Module -ListAvailable -Name WebAdministration))
     {
-        Throw 'Please ensure that WebAdministration module is installed.'
+        throw 'Please ensure that WebAdministration module is installed.'
     }
 
     $Website = Get-Website | Where-Object -FilterScript {
         $_.Name -eq $Name
     }
 
-    if ($Website.count -eq 0) # No Website exists with this name.
+    if ($Website.Count -eq 0) # No Website exists with this name.
     {
-        $ensureResult = 'Absent'
+        $EnsureResult = 'Absent'
     }
     elseif ($Website.count -eq 1) # A single Website exists with this name.
     {
-        $ensureResult = 'Present'
-
-        [PSObject[]] $Bindings
-        $Bindings = (Get-ItemProperty -Path IIS:\Sites\$Name -Name Bindings).collection
+        $EnsureResult = 'Present'
 
-        $CimBindings = foreach ($binding in $Bindings)
-        {
-            $BindingObject = Get-WebBindingObject -BindingInfo $binding
-            New-CimInstance -ClassName MSFT_xWebBindingInformation -Namespace root/microsoft/Windows/DesiredStateConfiguration -Property @{
-                Port                  = [System.UInt16]$BindingObject.Port
-                Protocol              = $BindingObject.Protocol
-                IPAddress             = $BindingObject.IPaddress
-                HostName              = $BindingObject.Hostname
-                CertificateThumbprint = $BindingObject.CertificateThumbprint
-                CertificateStoreName  = $BindingObject.CertificateStoreName
-                SSLFlags              = $BindingObject.SSLFlags
-            } -ClientOnly
-        }
+        $CimBindings = @(
+            $Website.bindings.Collection |
```

Comment:
Piping really isn't getting us anything here.

Could we just do:
````powershell
$CimBindings = @(ConvertTo-CimBinding -InputObject $Website.Bindings.Collection)
`````

Easier to read and should be faster.